File: packages/alphax_native/ios/Classes/AlphaXProgressInterest.swift
import Foundation

/// Operation-local progress interest; never shared between requests.
internal struct AlphaXProgressInterest {
    let downloadRequested: Bool
    let uploadRequested: Bool

    init(arguments: [String: Any]) {
        downloadRequested = arguments["downloadProgressRequested"] as? Bool ?? false
        uploadRequested = arguments["uploadProgressRequested"] as? Bool ?? false
    }

    func isRequested(direction: String) -> Bool {
        switch direction {
        case "download": return downloadRequested
        case "upload": return uploadRequested
        default: return false
        }
    }
}

File: packages/alphax_native/android/src/main/kotlin/com/auvana/ventures/alphax/AlphaXProgressInterest.kt
package com.auvana.ventures.alphax

/** Operation-local progress interest; never shared between requests. */
internal data class AlphaXProgressInterest(
    val downloadRequested: Boolean,
    val uploadRequested: Boolean,
) {
    fun isRequested(direction: String): Boolean = when (direction) {
        "download" -> downloadRequested
        "upload" -> uploadRequested
        else -> false
    }

    companion object {
        fun from(arguments: Map<String, Any?>): AlphaXProgressInterest = AlphaXProgressInterest(
            downloadRequested = arguments["downloadProgressRequested"] == true,
            uploadRequested = arguments["uploadProgressRequested"] == true,
        )
    }
}
